### PR TITLE
Minimal service worker

### LINKED ISSUE #1217: Implement minimal Service Worker
**Is your feature request related to a problem? Please describe.**
We have concerns around invalidating old service workers in the event of a 'catastrophe scenario'; particularly for users who keep the BBC open in a tab they never close.

As a first step, we'd like to remove all 'functionality' (caching) from our service worker and keep it as simple as possible, so that we continue to have 'a service worker' and all the routing required, etc, but it doesn't actually do anything useful (and therefore nothing potentially hazardous!)

Depended on by #1230 (service worker logging).

For context, we will set a low cache on sw.js in [RESFRAME-2561](https://jira.dev.bbc.co.uk/browse/RESFRAME-2561) (which also mitigates the catastrophe scenario).

**Describe the solution you'd like**
A PR to remove the Offline Plugin from Webpack and add a basic `sw.js` instead (perhaps under `public/`?) ([Some common service worker setups available here](https://serviceworke.rs)).

**Describe alternatives you've considered**
N/A

**Testing notes**
Test to make sure that the service worker still appears. Agree with Dev insight in terms of making sure we refresh the page and check the behaviour within the network tab.

Dev insight: service worker should continue to download on local/test/live, but won't do any caching of JS assets (vendor bundle, main bundle etc). When you refresh the page (gotcha: with caching enabled) the asset in the network tab should NOT say "served from service worker" (or equivalent).

**Additional context**
Full write-up here: https://paper.dropbox.com/doc/Service-worker-usage-investigation--AWkWauJ9fV7yqj7~FfZtpBpMAg-2RmFsBBOBLjzQS9oRUhbo

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #1217

**Overall change:** Implements a very minimal service worker

**Code changes:**

- Remove offline plugin
- Adds very minimal service worker

---
Testing: When `npm run build && npm start` the service worker should be loaded on news and persian. It should log when its installed and active. if you look at the 'verbose' console log it should console log when it sees files being loaded, but not act on it. 

There should be no service worker caching in the network tab 

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: public/sw.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
+var version = 'v0.1';
```
Comment (ChrisBAshton) [inline public/sw.js]: Can we make this match the version on `package.json`? For sanity reasons mainly.

File: public/sw.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+self.addEventListener("install", function(event) {
```
Comment (ChrisBAshton) [inline public/sw.js]:
```suggestion
self.addEventListener('install', function (event) {
```

ChrisLint™

File: public/sw.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  console.log('SW ' + version + ': install event.');
+});
+
+self.addEventListener("fetch", function(event) {
```
Comment (ChrisBAshton) [inline public/sw.js]:
```suggestion
self.addEventListener('fetch', function (event) {
```

ChrisLint™

File: public/sw.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  console.debug('SW ' + version + ': fetch event for ' + event.request.url);
+});
+
+self.addEventListener("activate", function(event) {
```
Comment (ChrisBAshton) [inline public/sw.js]:
```suggestion
self.addEventListener('activate', function (event) {
```

ChrisLint™

File: src/server/index.jsx [hunk 5]
```diff
@@ -111,7 +111,7 @@ server
     res.sendStatus(200);
   })
   .get(swRegexPath, (req, res) => {
-    const swPath = `${__dirname}/public/sw-${process.env.APP_ENV}.js`;
+    const swPath = `${__dirname}/public/sw.js`;
```
Comment (ChrisBAshton) [inline src/server/index.jsx:115]: Does this affect any of our Mozart issues / JIRA tickets / tests etc...?

**Review by ChrisBAshton — COMMENTED** [review]
Nice tests 👍 couple of comments

File: src/server/index.jsx (hunk 5 above)
Comment (dr3) [reply src/server/index.jsx:115]: Nope, this is purely internal, Mozart will continue to route to 'sw.js' in simorgh

**Review by pjlee11 — COMMENTED** [review]
Great work! 👍

**Review by pjlee11 — DISMISSED** [review]

File: public/sw.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
```
Comment (sareh) [inline public/sw.js]: The file at location `/${data.service}/articles/sw.js` is being registered in the `client.js`. This file should be moved into the relevant `/public/news/` directory.

Comment (sareh) [inline public/sw.js:1]: I'd suggest also moving this file to be in `public/news/sw.js` and updating this line to pull it from the service-specific directory: https://github.com/bbc/simorgh/blob/latest/src/server/index.jsx#L114
This is to ensure that we keep our service-workers for different BBC services clearly separated.

**Review by sareh — APPROVED** [review]
After having a conversation about this with Drew, we'll add the service-specific SW files in a future PR.

**Review by pjlee11 — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
👍

**Pull request merged**